### ISSUE TITLE
Distributed computing

### ISSUE THREAD

**Issue by riyijiye** [vrenkens/nabu#23]
Hi Vincent,

for multi-server multi-card distributed computing, it looks to me what you implemented here is much more complicated than the example shown in Tensorflow official website (local_cluster.py, create_server.py, ... ,)I am sure I missed many details, so may I know what you really achieve beyond the example provided by distributed Tensorflow?

thanks,
Harry

**Comment by vrenkens** [comment]
Hi Harry,

What I implemented adds no functionality but the standard stuff does not work in my department :). If the standard stuff works for your setup then it is much easier to use then what I implemented. 

The problem I faced when working on multiple machines in our department was that all network ports except the ssh port are closed for security reasons. I could not communicate through them, so the system failed. My solution was to use port forwarding over ssh. Nabu will first create links over ssh between all machine pairs in the cluster in two directions and then will build a cluster on top of those forwarded ports. 

This makes it all much more complicated then the standard stuff. I don't really know if this is the best way of doing it, but after a long time of struggling it is the best way I came up with :).
